app/services/katello/pulp3/repository.rb

```diff
@@ -116,12 +116,14 @@ def get_distribution(href)
       end
 
       def update_distributions
-        paths.values.select { |prefix, _path| prefix == 'http' }.each do |path|
```

Comment:
I realized this update method is looking quite a bit like the refresh_distributions method (which was meant to handle create/update iirc):
https://github.com/Katello/katello/blob/b06c0359c154951baea61be7c882e5aabbe49d02/app/services/katello/pulp3/repository.rb#L81-L89

would you want to just modify that to handle the deletion too?   One issue with the logic you have is that #paths won't return the http path if the repositories was set to 'unprotected' to be false.  We could let #paths always return both http and https, and let the calling method decide whether or not to create or delete the 'http' path.

Reply:
@jlsherrill yeah I went through of couple of iterations trying to get this just right earlier today. I was about to contemplate yet another attempt. I'll try modifying paths to include both http and https prefixes. 

I'll also look at the refresh_distributions. Perhaps there' s a compact and understandable way to modify that for the updates as well.